src/execution/index/art/art.cpp

```diff
@@ -646,10 +658,10 @@ bool ART::SearchEqual(ARTIndexScanState *state, idx_t max_count, vector<row_t> &
 	return true;
 }
 
-void ART::SearchEqualJoinNoFetch(Value &equal_value, idx_t &result_size) {
-	//! We need to look for a leaf
-	auto key = CreateKey(*this, types[0], equal_value);
-	auto leaf = (Leaf *)(Lookup(tree, *key, 0));
+void ART::SearchEqualJoinNoFetch(Key &key, idx_t &result_size) {
+
+	// we need to look for a leaf
+	auto leaf = (Leaf *)(Lookup(tree, key, 0));
```

Comment:
For what, exactly?

Reply:
Well lookup can be anything, but there is an assumption it's a leaf right

Reply:
Looking at the implementation, I think it can only be a `Leaf` or a `nullptr`? At least if the `NodeType` is set correctly?
```c++
Node *ART::Lookup(Node *node, Key &key, unsigned depth) {
	while (node) {
		if (node->type == NodeType::NLeaf) {
			auto leaf = (Leaf *)node;
			auto &leaf_prefix = leaf->prefix;
			//! Check leaf
			for (idx_t i = 0; i < leaf->prefix.Size(); i++) {
				if (leaf_prefix[i] != key[i + depth]) {
					return nullptr;
				}
			}
			return node;
		}
		if (node->prefix.Size()) {
			for (idx_t pos = 0; pos < node->prefix.Size(); pos++) {
				if (key[depth + pos] != node->prefix[pos]) {
					return nullptr;
				}
			}
			depth += node->prefix.Size();
		}
		idx_t pos = node->GetChildPos(key[depth]);
		if (pos == DConstants::INVALID_INDEX) {
			return nullptr;
		}
		node = node->GetChild(*this, pos);
		D_ASSERT(node);
		depth++;
	}
	return nullptr;
}
```

Reply:
Ah sure, I can improve the existing implementation to do that ;-) While I am on it I'll also change the `unsigned` type there 😋